entitybroker/core-providers/src/java/org/sakaiproject/entitybroker/providers/MembershipEntityProvider.java

```diff
@@ -847,6 +872,17 @@ public String createEntity(EntityReference ref, Object entity, Map<String, Objec
                     sg.site.addMember(userIds[i], roleId, active, false);
                     saveSiteMembership(sg.site);
                 }
+                User user = null;
+                // Add change to user_audits_log table.
+                try {
+                    user = userDirectoryService.getUser(userIds[i]);
+                }
+                catch (UserNotDefinedException e) {
+                    log.error(".createEntity: User with id {} doesn't exist", userIds[i]);
```

Comment:
Don't you want to stop here as if you continue you're going to get a NPE when building the `userAuditString`.